modules/mono/SCsub

```diff
@@ -27,3 +30,41 @@ if env["platform"] in ["macos", "ios"]:
 if env.editor_build:
     env_mono.add_source_files(env.modules_sources, "editor/*.cpp")
     SConscript("editor/script_templates/SCsub")
+
+    if env["build_csharp"]:
+        env["build_assemblies"] = get_cmdline_bool("build_assemblies", env["build_csharp"])
+        env["generate_mono_glue"] = get_cmdline_bool("generate_mono_glue", env["build_csharp"])
+
+    if env["build_assemblies"] or env["generate_mono_glue"]:
+        gd = os.path.join(env.Dir("#bin").abspath, "godot" + env["PROGSUFFIX"])
+
+        if env["generate_mono_glue"]:
+            env["BUILDERS"]["CSharpGlue"] = Builder(action=gd + " --headless --generate-mono-glue modules/mono/glue")
+
+            # the only generated files with fixed names are these
+            glue_targets = [
```

Comment:
A scons build rule requires a list of sources and a list of targets. This is so that the build system can figure out what build targets need to run when files are changed. In the bindings generation case, we have a problem: we don't exactly know what files contribute to bindings classes, because the binding generation data is calculated while it's running, and there's no metadata information that scons can read that says "this .h/.cpp file is going to generate a .cs file", short of literally regexing every file in the codebase looking for binding information (which, btw, can be done). Therefore we don't have an accurate list of target files to give to scons. We have to give it something, and there are files that are always a part of the code generation, so we can give scons this list. This list doesn't have to be super accurate, because no other build rules rely on this rule for their dependency tracking.

Reply:
I see, my concerns are that this list may get outdated so it's one more thing we'll have to keep track of.

Reply:
I'm hoping at least the `NativeFuncs.cs` file will always exist? We only really need one file to give to scons here, so whatever file we are very unlikely to rename will do. It needs to be a file that is generated by the generator, so that if it's missing scons knows it needs to rerun this rule.

We could record the filenames that are generated into a text file, commit that to the repo, and use that as the initial source of files to track, instead of hardcoding a file list here. That way it would be more accurate and wouldn't go stale.

The ideal ideal solution, I think, would be to have a dry-run generator pass that would output the list of .h/.cpp sources and the list of .cs targets relevant to the bindings generator, without doing anything else. We would run that as build step right after the c++ build, and then this rule would have exactly the right list of sources and targets to know when it would need to run.

Reply:
Can't we use a globbing pattern here, even if may encompass more files than absolutely necessary?

As a first step we probably want to be conservative and it's better to match more files than needed, than to miss some.

Reply:
I think you meant `NativeCalls.cs`. `NativeFuncs.cs` is not generated.

I think the file that is least likely to ever be renamed/moved/removed is probably `GeneratedIncludes.props`, but to be honest I think it's not very likely that any of the files you list here will be.

What you describe as the ideal solution makes sense to me, but maybe we don't need to make it that complicated. Every generated file is always re-generated when generating the glue (even if the generated file would have the same content), so I'm assuming that means SCons will think the file changed. So I don't really see the benefits of this over the current implementation, but maybe I'm missing something.